std/math.d

```diff
@@ -6226,7 +6226,53 @@ public:
  *      x =     the value to evaluate.
  *      A =     array of coefficients $(SUB a, 0), $(SUB a, 1), etc.
  */
-real poly(real x, const real[] A) @trusted pure nothrow @nogc
+Unqual!(CommonType!(T1, T2)) poly(T1, T2)(T1 x, in T2[] A) @trusted pure nothrow @nogc
+    if (isFloatingPoint!T1 && isFloatingPoint!T2)
+in
+{
+    assert(A.length > 0);
+}
+body
+{
+    static if(is(Unqual!T2 == real))
+    {
+        return polyImpl(x, A);
+    }
+    else
+    {
+        ptrdiff_t i = A.length - 1;
+        typeof(return) r = A[i];
+        while (--i >= 0)
```

Comment:
I'm uneasy about this. While it's unlikely we will ever hit this case, this loop condition will fail if `A.length-1` is too large to fit in a `ptrdiff_t`. Why not write it in a more watertight way?
````
size_t i = A.length - 1; // OK, because the in-contract asserts length > 0
typeof(return) r = A[i];
while (i > 0)
{
    r *= x;
    r += A[i];
    --i;
}
````